src/main/scala/cognite/spark/v1/RelationshipsRelation.scala

```diff
@@ -0,0 +1,156 @@
+      }
+
+    // Merge streams related to each partition to make sure duplicate values are read into
+    // the same RDD partition
+    streamsPerFilter.transpose
+      .map(s => s.reduce(_.merge(_)))
+  }
+
+  def relationshipsFilterFromMap(m: Map[String, String]): RelationshipsFilter =
+    RelationshipsFilter(
+      sourceExternalIds = m.get("sourceExternalIds").map(stringSeqFromWrappedArray),
+      sourceTypes = m.get("sourceTypes").map(stringSeqFromWrappedArray),
+      targetExternalIds = m.get("targetExternalIds").map(stringSeqFromWrappedArray),
+      targetTypes = m.get("targetTypes").map(stringSeqFromWrappedArray),
+      dataSetIds = m.get("dataSetId").map(idsFromWrappedArray(_).map(CogniteInternalId)),
+      startTime = timeRangeFromMinAndMax(m.get("minStartTime"), m.get("maxStartTime")),
+      endTime = timeRangeFromMinAndMax(m.get("minEndTime"), m.get("maxEndTime")),
+      activeAtTime = timeRangeFromMinAndMax(m.get("minActiveAtTime"), m.get("maxActiveAtTime")),
+      labels = getLabelsFilter(m.get("labels")),
+      confidence = confidenceRangeFromMinAndMax(m.get("minConfidence"), m.get("maxConfidence")),
+      createdTime = timeRangeFromMinAndMax(m.get("minCreatedTime"), m.get("maxCreatedTime")),
+      lastUpdatedTime = timeRangeFromMinAndMax(m.get("minLastUpdatedTime"), m.get("maxLastUpdatedTime"))
+    )
+
+  override def insert(rows: Seq[Row]): IO[Unit] = {
+    val relationships = rows.map(fromRow[RelationshipCreate](_))
+    client.relationships
+      .create(relationships)
+      .flatTap(_ => incMetrics(itemsCreated, relationships.length)) *> IO.unit
+  }
+
+  override def delete(rows: Seq[Row]): IO[Unit] = {
+    val relationshipIds = rows.map(fromRow[RelationshipsDeleteSchema](_)).map(_.externalId)
+    client.relationships
+      .deleteByExternalIds(relationshipIds)
+      .flatTap(_ => incMetrics(itemsDeleted, relationshipIds.length))
+  }
+
+  override def upsert(rows: Seq[Row]): IO[Unit] =
+    throw new CdfSparkException("Upsert is not supported for relationships")
+
+  override def update(rows: Seq[Row]): IO[Unit] =
+    throw new CdfSparkException("Update is not supported for relationships")
+}
+
+object RelationshipsRelation {
+  var insertSchema: StructType = structType[RelationshipsInsertSchema]
+  var readSchema: StructType = structType[RelationshipsReadSchema]
+  var deleteSchema: StructType = structType[RelationshipsDeleteSchema]
+}
+
+final case class RelationshipsDeleteSchema(
+    externalId: String
+)
+
+final case class RelationshipsInsertSchema(
+    externalId: Option[String] = None,
+    sourceExternalId: String,
+    sourceType: String,
+    targetExternalId: String,
+    targetType: String,
+    startTime: Option[Instant] = None,
+    endTime: Option[Instant] = None,
+    confidence: Float = 0,
+    labels: Option[Seq[Map[String, String]]] = None,
+    dataSetId: Option[Long] = None
+)
+
+final case class RelationshipsReadSchema(
+    externalId: Option[String] = None,
```

Comment:
@hakontro Is this also used for update/upsert?

Reply:
This is only there so we can expose it to Transformations actually, so that it can know what the column schema of reading a Relationship is like

Reply:
Again, labels give me trouble here :D It is defined as a Option[Seq[CogniteExternalId]] in scala-sdk. Should I use the same type or keep the version here?

Reply:
Did we figure this one out? Don't remember.

Reply:
I solved it by using RelationshipsReadSchema instead of Relationship class. I converted labels into a list of strings.

Reply:
It would be great if you can review from start since it is changed a bit.